Mathlib/RingTheory/Ideal/IsPrimary.lean

```diff
@@ -14,33 +15,52 @@ A proper ideal `I` is primary iff `xy ∈ I` implies `x ∈ I` or `y ∈ radical
 
 - `Ideal.IsPrimary`
 
+## Implementation details
+
+Uses a specialized phrasing of `Submodule.IsPrimary` to have better API-piercing usage.
+
 -/
 
 namespace Ideal
 
 variable {R : Type*} [CommSemiring R]
 
+/-- A proper ideal `I` is primary as a submodule. -/
+abbrev IsPrimary (I : Ideal R) : Prop :=
+  Submodule.IsPrimary I
+
+/-- An ideal `I : Ideal R` is primary iff it is primary as a submodule. This links
+`Ideal.IsPrimary` with `Submodule.IsPrimary`. -/
+lemma isPrimary_iff {I : Ideal R} :
+    I.IsPrimary ↔ Submodule.IsPrimary I :=
+  Iff.rfl
```

Comment:
I consider this unnecessary since you can `rw [Ideal.IsPrimary]`. Maybe give its name to `isPrimary_iff'`?

Reply:
I was following @jcommelin's suggestion. Thoughts?

Reply:
I guess it's fine to drop this. What I mainly cared about was aligning the defeq with `Submodule.IsPrimary`.